src/SonarScanner.MSBuild.Shim/SonarEngineWrapper.cs

```diff
@@ -54,4 +54,35 @@ public virtual bool Execute(AnalysisConfig config, string standardInput)
         }
         return result.Succeeded;
     }
+
+    private string FindJavaExe(string configJavaExe)
```

Comment:
Not a requirement, but I would be a fan of restructuring this method to look something like this:
```suggestion
    private string FindJavaExe(string configJavaExe) =>
        JavaFromConfig(...)
        ?? JavaFromJavaHome(...)
        ?? JavaFromPath(...)
```